src/twisted/trial/_dist/functional.py

```diff
@@ -28,6 +28,18 @@ def fromOptional(default: _A, optional: Optional[_A]) -> _A:
     return optional
 
 
+def takeWhile(condition: Callable[[_A], bool], xs: Iterable[_A]) -> Iterable[_A]:
+    """
+    :return: An iterable of the prefix of ``xs`` until ``condition`` returns
+        ``False``.
```

Comment:
Maybe  rename it 'iterateWhile` or `iterWhile` ?

```suggestion
    :return: An iterable over C{xs}} that stops when C{condition} returns
        ``False`` based on the value of iterated C{xs}.
```

Reply:
I'd like to leave the name for now.  It's directly copied from Haskell - https://hackage.haskell.org/package/base-4.16.1.0/docs/Prelude.html#v:takeWhile - and I think it might make sense to maintain that consistency.  If not, this is all private, I won't mind renaming stuff later if this proves to be a bad idea.